### DISCUSSION TITLE
Cross-Domain iframe Storage Limits

### DISCUSSION THREAD

**Discussion by iamnewton** [MagniteEngineering/fledge.polyfill#6] [Specification]
Given we’re going to use 1 domain/iframe as the mock dB storage for all of the interest groups that a user joins, do we need to be concerned about any storage limitations that the browser may enforce?

**Comment by brodrigu** [comment]
Yes, assuming such limits exist in chrome (the only browser we can support).

**Reply by iamnewton** [reply]
According to https://web.dev/storage-for-the-web/#how-much, Chrome will allow us up to 80% of total disk space.  We can use the [StorageManager API](https://developer.mozilla.org/en-US/docs/Web/API/StorageManager/estimate) to test for its limits.

```javascript
if (navigator.storage && navigator.storage.estimate) {
  const quota = await navigator.storage.estimate();
  // quota.usage -> Number of bytes used.
  // quota.quota -> Maximum number of bytes available.
  const percentageUsed = (quota.usage / quota.quota) * 100;
  console.log(`You've used ${percentageUsed}% of the available storage.`);
  const remaining = quota.quota - quota.usage;
  console.log(`You can write up to ${remaining} more bytes.`);
}
```
